test/SimpleWebRequest.spec.ts

```diff
@@ -161,154 +172,199 @@ describe('SimpleWebRequest', () => {
             const onSuccessCritical2 = jasmine.createSpy('onSuccessCritical2');
             const status = 200;
 
-            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.Low }).start().then(onSuccessLow1);
+            const p1 = new SimpleWebRequest<string>(method, url, { priority: WebRequestPriority.Low })
+                .start().then(onSuccessLow1);
             jasmine.clock().tick(10);
 
-            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.Critical }).start().then(onSuccessCritical1);
+            const p2 = new SimpleWebRequest<string>(method, url, { priority: WebRequestPriority.Critical })
+                .start().then(onSuccessCritical1);
             jasmine.clock().tick(10);
 
-            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.Low }).start().then(onSuccessLow2);
+            const p3 = new SimpleWebRequest<string>(method, url, { priority: WebRequestPriority.Low })
+                .start().then(onSuccessLow2);
             jasmine.clock().tick(10);
 
             SimpleWebRequestOptions.MaxSimultaneousRequests = 1;
             // add a new request to kick the queue
-            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.Critical }).start().then(onSuccessCritical2);
-
-            // only one is executed
-            expect(jasmine.Ajax.requests.count()).toBe(1);
-            jasmine.Ajax.requests.mostRecent().respondWith({status});
-            // they're executed in correct order
-            expect(onSuccessCritical1).toHaveBeenCalled();
-
-            jasmine.Ajax.requests.mostRecent().respondWith({status});
-            expect(onSuccessCritical2).toHaveBeenCalled();
-
-            jasmine.Ajax.requests.mostRecent().respondWith({status});
-            expect(onSuccessLow1).toHaveBeenCalled();
-
-            jasmine.Ajax.requests.mostRecent().respondWith({status});
-            expect(onSuccessLow2).toHaveBeenCalled();
+            const p4 = new SimpleWebRequest<string>(method, url, { priority: WebRequestPriority.Critical })
+                .start().then(onSuccessCritical2);
+
+            asyncTick().then(() => {
+                // only one is executed
+                expect(jasmine.Ajax.requests.count()).toBe(1);
```

Comment:
Unrelated to your change; should each `.then` check `count toBe n` or something similar? After the first response, then all of them could execute at the same time and the test would be meaningless.

[Same for `'after the request is unblocked, it\'s returned to the queue with correct priority'` and others that expect a certain order.]